platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginExtension.java

```diff
@@ -270,73 +269,17 @@ private static String validateFeatureName(String name) {
         return name;
     }
 
+    private boolean isJavaComponentPresent() {
+        return null != project.getComponents().findByName(JvmConstants.JAVA_COMPONENT_NAME);
```

Comment:
Should probably test for _any_ component with type `JvmSoftwareComponentInternal`, instead of testing by name, for consistency with above code